src/shared/wizards/wizardForm.ts

```diff
@@ -3,87 +3,159 @@
  * SPDX-License-Identifier: Apache-2.0
```

Comment:
As a whole, within this file, there is a LOT of logic running through types, which are as far as I know untestable. Is it worth investing this heavily in types? The syntax is obscure, and troubleshooting/editing/general maintenance is not going to be easy, especially in the case of people rotating in/out of the team with less Typescript knowledge.

As a thought experiment: are there other ways to implement this that relies less on the type system and more on testable helper functions? I might be missing something, but it doesn't seem like there's any way to test the output of the toposort, which seems to play heavily into the types here (I very well might be wrong)?

Reply:
Well this file is like 90% documentation now. The types here are really not that crazy, it just seems weird if you have never dealt with the syntax at all. Will be sitting on this PR until after the next release though. I'll see what I can part out in the meantime to make it easier to digest.

Reply:
It has been a while since I have contributed to this project, but I want to make a quick comment on this because it seems like a fundamental change in the style of the code base. 

Originally, the toolkit for VSCode used the Typescript type system much more fully than it does today. However, after the people working on it changed significantly, a lot of the uses of the more advanced type system features were removed because they made the code hard to understand. Personally, I feel like I have a good handle on generics and some handle on template metaprogramming in C++, but I find a lot of this code is hard to parse: I'd rather try to achieve the same level of type safety with duplication and more verbosity.

If you believe this code will not require significant changes or debugging in the future, dense code like this is certainly less of a problem. However, in my experience, there is not that much code that does not change in an actively developing and expanding project like this, especially as VSCode itself changes.